app/tests/spec/views/permissions.js

```diff
@@ -101,107 +115,272 @@ define(function (require, exports, module) {
-              'shows email in permissions list');
+          it('redirects to /signup', function () {
+            assert.isTrue(view.navigate.calledWith('/signup'));
           });
+        });
       });
     });
 
     describe('submit', function () {
       beforeEach(function () {
         sinon.spy(account, 'saveGrantedPermissions');
-        sinon.spy(user, 'setAccount');
       });
 
-      it('coming from sign in, redirects unverified users to the confirm page on success', function () {
-        return initView('sign_in')
-          .then(function () {
+      describe('with unverified users from sign in', function () {
+        beforeEach(function () {
+          account.unset('verified');
+          return initView('sign_in')
+            .then(function () {
 
-            return view.submit()
-              .then(function () {
-                assert.isTrue(account.saveGrantedPermissions.calledWith(CLIENT_ID, PERMISSIONS));
-                assert.isTrue(user.setAccount.calledWith(account));
-                assert.isTrue(view.navigate.calledWith('confirm', {
-                  account: account
-                }));
-              });
-          });
+              sinon.spy(view, 'onSignInUnverified');
+
+              return view.submit();
+            });
+        });
+
+        it('delegates to `onSignInUnverified`', function () {
+          assert.isTrue(account.saveGrantedPermissions.calledWith(CLIENT_ID, PERMISSIONS));
+          assert.isTrue(user.setAccount.calledWith(account));
+          assert.isTrue(view.onSignInUnverified.calledWith(account));
+        });
+      });
+
+      describe('with verified users from sign in', function () {
+        beforeEach(function () {
+          account.set('verified', true);
+          return initView('sign_in')
+            .then(function () {
+
+              sinon.spy(view, 'onSignInSuccess');
+
+              return view.submit();
+            });
+        });
+
+        it('delegates to `onSignInSuccess`', function () {
+          assert.isTrue(account.saveGrantedPermissions.calledWith(CLIENT_ID, PERMISSIONS));
+          assert.isTrue(user.setAccount.calledWith(account));
+          assert.isTrue(view.onSignInSuccess.calledWith(account));
+          assert.isTrue(TestHelpers.isEventLogged(metrics,
+                            'permissions.success'));
+        });
+      });
+
+      describe('with users from sign up', function () {
+        beforeEach(function () {
+          return initView('sign_up')
+            .then(function () {
+
+              sinon.spy(view, 'onSignUpSuccess');
+
+              return view.submit();
+            });
+        });
+
+        it('delegates to `onSignUpSuccess`', function () {
+          assert.isTrue(account.saveGrantedPermissions.calledWith(CLIENT_ID, PERMISSIONS));
+          assert.isTrue(user.setAccount.calledWith(account));
+          assert.isTrue(view.onSignUpSuccess.calledWith(account));
```

Comment:
It does. This whole section is going to go away with #3413, but a good idea to add a test until that is merged.